Doc/whatsnew/3.11.rst

```diff
@@ -1173,6 +1173,61 @@ Deprecated
   (Contributed by Serhiy Storchaka in :gh:`91760`.)
 
 
+Pending Removal in Python 3.12
+==============================
+
+The following APIs have been deprecated in earlier Python releases, and are now
+pending removal in Python 3.12.
+
+Python API:
+
+* :class:`pkgutil.ImpImporter`
+* :class:`pkgutil.ImpLoader`
+* :envvar:`PYTHONTHREADDEBUG`
+* :func:`importlib.find_loader`
+* :func:`importlib.util.module_for_loader`
+* :func:`importlib.util.set_loader_wrapper`
+* :func:`importlib.util.set_package_wrapper`
+* :meth:`importlib.abc.Loader.module_repr`
+* :meth:`importlib.abc.Loadermodule_repr`
+* :meth:`importlib.abc.MetaPathFinder.find_module`
+* :meth:`importlib.abc.MetaPathFinder.find_module`
+* :meth:`importlib.abc.PathEntryFinder.find_loader`
+* :meth:`importlib.abc.PathEntryFinder.find_module`
+* :meth:`importlib.machinery.BuiltinImporter.find_module`
+* :meth:`importlib.machinery.BuiltinLoader.module_repr`
+* :meth:`importlib.machinery.FileFinder.find_loader`
+* :meth:`importlib.machinery.FileFinder.find_module`
+* :meth:`importlib.machinery.FrozenImporter.find_module`
+* :meth:`importlib.machinery.FrozenLoader.module_repr`
+* :meth:`importlib.machinery.PathFinder.find_module`
+* :meth:`importlib.machinery.WindowsRegistryFinder.find_module`
+* :meth:`pathlib.Path.link_to`
+* The entire :ref:`distutils namespace <distutils-deprecated>`
+* ``cgi.log()``
+* ``sqlite3.OptimizedUnicode``
+* ``sqlite3.enable_shared_cache``
```

Comment:
I agree with the consistency argument, but does Sphinx still generate (null/404) links on role targets that don't exist in the index?

Reply:
It generates a non-clickable black text, instead of a blue one.  Note that ` ``...`` ` also adds a darker background, and `` :func:`...` `` adds `()` at the end.  Sphinx also doesn't raise any warning for invalid targets, and neither does `make check` so we don't have to worry about that (maybe they should though?).

IOW, I agree with what @vstinner suggested.